src/core/observer/index.js

```diff
@@ -188,6 +188,13 @@ export function defineReactive (
  * already exist.
  */
 export function set (obj: Array<any> | Object, key: any, val: any) {
```

Comment:
Yes, but to follow JS's standard, objects passed in as key are coerced to strings.